### PR TITLE
Add support for blob length for different dialects

### PR DESCRIPTION
**Pull request by pgghosh**
Fixed bloblength function to use the correct db function based on the dialect.
Also added support for blob literal value evaluation using the appropriate DB function.

Changes made to MySQL, Oracle and PostGreSQL dialects only as these are the only supported dialects as of now.

Blob literals store binary data in HEX format as most DBs support saving data to DB columns in HEX format.

### CODE REVIEW

File: morf-core/src/main/java/org/alfasoftware/morf/sql/element/BlobFieldLiteral.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Since we store binary data, we will store the values in hex format, since all DBs support inserting
+ * blob data using hex values.
+ * So {@link FieldLiteral#getValue()} will always return the hex encoded string.
+ */
+public class BlobFieldLiteral extends FieldLiteral {
+
+    public BlobFieldLiteral(byte[] bytes) {
+        super("", new BigInteger(bytes).toString(16).toUpperCase(), DataType.BLOB);
```
Comment (capgen628) [inline morf-core/src/main/java/org/alfasoftware/morf/sql/element/BlobFieldLiteral.java:16]: There's another super constructor that doesn't take the alias

File: morf-core/src/main/java/org/alfasoftware/morf/sql/SqlUtils.java [hunk 2]
```diff
@@ -343,6 +344,28 @@ public static FieldLiteral literal(Character value) {
   }
 
 
+  /**
+   * Constructs a new {@link BlobFieldLiteral} with a byte array source.
+   *
+   * @param value the literal value to use
+   * @return {@link BlobFieldLiteral}
+   */
+  public static BlobFieldLiteral blobliteral(byte[] value) {
```
Comment (capgen628) [inline morf-core/src/main/java/org/alfasoftware/morf/sql/SqlUtils.java:353]: Could we just call this method blob?

File: morf-core/src/main/java/org/alfasoftware/morf/sql/SqlUtils.java [hunk 3]
```diff
@@ -343,6 +344,28 @@ public static FieldLiteral literal(Character value) {
...
+  }
+
+
+  /**
+   * Constructs a new {@link BlobFieldLiteral} with a String source.
+   *
+   * @param value the literal value to use
+   * @return {@link BlobFieldLiteral}
+   */
+  public static BlobFieldLiteral blobliteral(String value) {
```
Comment (capgen628) [inline morf-core/src/main/java/org/alfasoftware/morf/sql/SqlUtils.java:364]: Could we just call this method blob?

File: morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlDialect.java [hunk 4]
```diff
@@ -1966,6 +1965,12 @@ protected String getSqlFrom(Function function) {
     }
   }
 
+  private void checkFunctionHasSingleArgument(Function function) {
```
Comment (capgen628) [inline morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlDialect.java:1968]: checkSingleArgument?

**Review by capgen628 — CHANGES_REQUESTED** [review]

**Review by gilleain — APPROVED** [review]

**Comment by gilleain** [comment]
Yes, looks fine.

**Pull request merged**